### PR TITLE
request prepare route only on cloud5

### PR DESCRIPTION
**Pull request by MaximilianMeister**
this route does not exist (yet) in cloud6

### CODE REVIEW

**Comment by dirkmueller** [comment]
@MaximilianMeister should it rather be added to cloud6 ? or is it not needed for the cloud7 update?

**Comment by MaximilianMeister** [comment]
> should it rather be added to cloud6 ? or is it not needed for the cloud7 update?

I would like to see if it is needed while fixing the cloud6 to 7 upgrade path. for now it would be ok to use it only on cloud5.

**Comment by bmwiedemann** [comment]
+1

**Comment by MaximilianMeister** [comment]
prepare was executed in the gating (upgrade 5 to 6)

subsequent gating failure is the same as in the other cloud-mkcloud6-job-upgrade builds

**Comment by MaximilianMeister** [comment]
sorry for stressing but this blocks my work on the 6to7 upgrade a bit

@dirk @jdsn @rsalevsky can you please review?

**Comment by jsuchome** [comment]
Makes sense. +1

**Comment by vuntz** [comment]
LGTM, but would prefer to have @jdsn give a +1 (to make sure it's the right way forward)

**Comment by MaximilianMeister** [comment]
> LGTM, but would prefer to have @jdsn give a +1 (to make sure it's the right way forward)

It is more a crowbar question i would say, as we don't have this route (yet) in cloud6. in fact this route only exists on cloud 5 and gets executed for cloud 6 and 7.

but thanks for the input. I will ping @jdsn

File: scripts/qa_crowbarsetup.sh [hunk 1]
```diff
@@ -4016,9 +4016,11 @@ function onadmin_prepare_crowbar_upgrade()
     if iscloudver 4minus ; then
         complain 11 "This upgrade path is only supported for Cloud 5+"
     else
-        # using the API, due to missing crowbar cli integration
-        # move nodes to upgrade mode
-        safely curl -s -X POST $crowbar_api_digest $crowbar_api/installer/upgrade/prepare
+        if iscloudver 5; then
```
Comment (bmwiedemann) [inline scripts/qa_crowbarsetup.sh]: could use elif here. or case $(getcloudver)

Comment (jdsn) [inline scripts/qa_crowbarsetup.sh]: +1 for a case $(getcloudver)

File: scripts/qa_crowbarsetup.sh [hunk 2]
```diff
@@ -4013,13 +4013,17 @@ function onadmin_cloudupgrade_reboot_and_redeploy_clients()
...
 {
-    if iscloudver 4minus ; then
-        complain 11 "This upgrade path is only supported for Cloud 5+"
-    else
-        # using the API, due to missing crowbar cli integration
-        # move nodes to upgrade mode
-        safely curl -s -X POST $crowbar_api_digest $crowbar_api/installer/upgrade/prepare
-    fi
+    case $(getcloudver) in
+        4minus)
```
Comment (jdsn) [inline scripts/qa_crowbarsetup.sh]: getcloudver only returns a number (not minus)

File: scripts/qa_crowbarsetup.sh [hunk 3]
```diff
@@ -4013,13 +4013,17 @@ function onadmin_cloudupgrade_reboot_and_redeploy_clients()
...
+            complain 11 "This upgrade path is only supported for Cloud 5+"
```
Comment (jdsn) [inline scripts/qa_crowbarsetup.sh]: Should be:

```
complain 11 "This upgrade path is only supported for Cloud 5"
```

File: scripts/qa_crowbarsetup.sh [hunk 4]
```diff
@@ -4013,13 +4013,17 @@ function onadmin_cloudupgrade_reboot_and_redeploy_clients()
...
+            ;;
+        5)
+            # using the API, due to missing crowbar cli integration
+            # move nodes to upgrade mode
+            safely curl -s -X POST $crowbar_api_digest $crowbar_api/installer/upgrade/prepare
+            ;;
+        # by default we skip this route as it doesn't exist for that cloud version
```
Comment (dirkmueller) [inline scripts/qa_crowbarsetup.sh]: maybe add a fat warning for the "6)" case that this needs to be implemented.. or rather commit a dummy implementation in crowbar (whatever moves us forward is fine..)

**Comment by MaximilianMeister** [comment]
I will rather add the route for cloud 6 now in crowbar, and then we will probably not need this anymore. thanks for the hint @dirkmueller

**Comment by MaximilianMeister** [comment]
even if it is not tested yet, https://github.com/crowbar/crowbar-core/pull/430 is the favoured way to go here

closing

**Pull request closed without merging**